src/App.tsx

```diff
@@ -9,42 +9,41 @@ import RecipeList from "./containers/RecipeList";
 import RecipeDetail from "./containers/RecipeDetail";
 import AddRecipeForm from "./components/AddRecipeForm";
 
-class App extends Component {
-  constructor(props) {
-    super(props);
+interface IAppState {
+  isAddingRecipe: boolean
+  selectedRecipe: boolean
+}
 
-    this.state = {
-      isAddingRecipe: false
-    };
+class App extends Component<{}, IAppState> {
```

Comment:
Note the `<{}, IAppState>`. The {} refers to the props (which this component doesn't have any so it's empty) and the `IAppState` is the local state for the component. This declaration here is how typescript knows what this.props and this.state should be when you reference either of those inside the component (and gives nice features like code completion)